packages/cms-cli/commands/lint.js

```diff
@@ -27,51 +27,76 @@ const {
-    .description('lint a file or folder for HubL syntax')
-    .arguments('<path>')
-    .action(async filepath => {
-      setLogLevel(command);
-      logDebugInfo(command);
-      const { config: configPath } = command;
-      loadConfig(configPath);
-      checkAndWarnGitInclusion();
+const loadAndValidateOptions = async options => {
+  setLogLevel(options);
+  logDebugInfo(options);
+  const { config: configPath } = options;
+  loadConfig(configPath);
+  checkAndWarnGitInclusion();
 
-      if (!(validateConfig() && (await validatePortal(command)))) {
-        process.exit(1);
-      }
+  if (!(validateConfig() && (await validatePortal(options)))) {
+    process.exit(1);
+  }
+};
 
-      const portalId = getPortalId(command);
+const action = async (args, options) => {
+  await loadAndValidateOptions(options);
 
-      trackCommandUsage(COMMAND_NAME, {}, portalId);
+  const portalId = getPortalId(options);
+  const localPath = resolveLocalPath(args.localPath);
+  const groupName = `Linting "${localPath}"`;
 
-      filepath = resolveLocalPath(filepath);
+  trackCommandUsage(COMMAND_NAME, {}, portalId);
 
-      const groupName = `Linting "${filepath}"`;
-      logger.group(groupName);
-      let count = 0;
-      try {
-        await lint(portalId, filepath, result => {
-          count += printHublValidationResult(result);
-        });
-      } catch (err) {
-        logger.groupEnd(groupName);
-        logErrorInstance(err, { portalId });
-        process.exit(1);
-      }
-      logger.groupEnd(groupName);
-      logger.log('%d issues found', count);
+  logger.group(groupName);
+  let count = 0;
+  try {
+    await lint(portalId, localPath, result => {
+      count += printHublValidationResult(result);
     });
-
-  addConfigOptions(command);
-  addPortalOptions(command);
-  addLoggerOptions(command);
-  addHelpUsageTracking(command, COMMAND_NAME);
-}
+  } catch (err) {
+    logger.groupEnd(groupName);
+    logErrorInstance(err, { portalId });
+    process.exit(1);
+  }
+  logger.groupEnd(groupName);
+  logger.log(`${count} issues found`);
+};
 
 module.exports = {
-  configureCommand,
+  /**
+   * Yargs
+   */
+  command: `${COMMAND_NAME} <path>`,
+  describe: DESCRIPTION,
+  builder: yargs => {
+    addConfigOptions(yargs, true);
+    addPortalOptions(yargs, true);
+    addLoggerOptions(yargs, true);
+    yargs.positional('path', {
+      describe: 'Local folder to lint',
+      type: 'string',
+    });
+    return yargs;
+  },
+  handler: async argv => action({ localPath: argv.path }, argv),
+  /**
+   * Commander
+   */
+  configureLintCommand: commander => {
+    commander
+      .version(version)
+      .description(DESCRIPTION)
+      .arguments('<path>')
+      .action(async (localPath, command = {}) =>
+        action({ localPath }, command)
+      );
+
+    addConfigOptions(commander);
+    addPortalOptions(commander);
+    addLoggerOptions(commander);
+    addHelpUsageTracking(commander, COMMAND_NAME);
+  },
```

Comment:
Sorry to nitpick so much, but I think having all this logic inside `module.exports` makes it harder to see at a glance what is being exported. I think that I didn't explain thoroughly how I thought using `module.exports` would work.
I was thinking something like:
```js
const command = `${COMMAND_NAME} <path>`;
const describe =  DESCRIPTION;
const builder = yargs => {
  addConfigOptions(yargs, true);
  addPortalOptions(yargs, true);
  addLoggerOptions(yargs, true);
  yargs.positional('path', {
    describe: 'Local folder to lint',
    type: 'string',
  });
  return yargs;
};
// etc…
module.exports = {
  // Yargs
  command,
  describe,
  builder,
  // Commander
  configureLintCommand,
};
```

Reply:
Okay, I'm not huge on that since it just mixes all of the configuration stuff in with functions that aren't getting exported, but I'll swap it

Reply:
Yeah, that is why `export.command` (or named ES6 exports) can be preferable.